apache2/re.c

```diff
@@ -111,23 +115,59 @@ static int fetch_target_exception(msre_rule *rule, modsec_rec *msr, msre_var *va
                     value = NULL;
                 }
 
+                value_len = 0;
+                if (value != NULL) {
+                    value_len = strlen(value);
+                }
+
                 if((strlen(myname) == strlen(name)) &&
-                        (strncasecmp(myname, name,strlen(myname)) == 0))   {
+                        (strncasecmp(myname, name, strlen(myname)) == 0))   {
 
                     if(value != NULL && myvalue != NULL)  {
-                        if((strlen(myvalue) == strlen(value)) &&
+                        if(value_len > 2 && value[0] == '/' && value[value_len - 1] == '/') {
+                            value[value_len - 1] = '\0';
+#ifdef WITH_PCRE2
+                            regex = msc_pregcomp(msr->mp, value + 1,
```

Comment:
This (copied) code contains a memory leak: the `regex` variable compiled in each cycle during the exclusion check, but never freed (never called [msc_pcre_cleanup()](https://github.com/owasp-modsecurity/ModSecurity/blob/v2/master/apache2/msc_pcre.c#L21-L49)).

Reply:
We should reorganize this part. I would expect that `regex` variable above must be compiled during startup, and not when the engine inspect the exclusion.